contrib/MedicalSeg/deploy/python/nnformer_infer.py

```diff
@@ -0,0 +1,514 @@
+        help='Print GLOG information of Paddle Inference.')
+
+    parser.add_argument(
+        '--use_swl',
+        default=False,
+        type=eval,
+        help='use sliding_window_inference')
+
+    parser.add_argument('--use_warmup', default=True, type=eval, help='warmup')
+
+    parser.add_argument('--img_shape', default=128, type=int, help='img_shape')
+
+    parser.add_argument('--is_nhwd', default=True, type=eval, help='is_nhwd')
+    return parser.parse_args()
+
+
+def use_auto_tune(args):
+    return hasattr(PredictConfig, "collect_shape_range_info") \
+           and hasattr(PredictConfig, "enable_tuned_tensorrt_dynamic_shape") \
+           and args.device == "gpu" and args.use_trt and args.enable_auto_tune
+
+
+class DeployConfig:
+    def __init__(self, path):
+        with codecs.open(path, 'r', 'utf-8') as file:
+            self.dic = yaml.load(file, Loader=yaml.FullLoader)
+
+        self._transforms = self.load_transforms(self.dic['Deploy'][
+            'transforms'])
+        self._dir = os.path.dirname(path)
+
+    @property
+    def transforms(self):
+        return self._transforms
+
+    @property
+    def model(self):
+        return os.path.join(self._dir, self.dic['Deploy']['model'])
+
+    @property
+    def params(self):
+        return os.path.join(self._dir, self.dic['Deploy']['params'])
+
+    @staticmethod
+    def load_transforms(t_list):
+        com = manager.TRANSFORMS
+        transforms = []
+        for t in t_list:
+            ctype = t.pop('type', None)
+            if ctype is not None:
+                transforms.append(com[ctype](**t))
+
+        return T.Compose(transforms)
+
+
+def auto_tune(args, imgs, img_nums):
+    """
+    Use images to auto tune the dynamic shape for trt sub graph.
+    The tuned shape saved in args.auto_tuned_shape_file.
+    Args:
+        args(dict): input args.
+        imgs(str, list[str]): the path for images.
+        img_nums(int): the nums of images used for auto tune.
+    Returns:
+        None
+    """
+    logger.info("Auto tune the dynamic shape for GPU TRT.")
+
+    assert use_auto_tune(args)
+
+    if not isinstance(imgs, (list, tuple)):
+        imgs = [imgs]
+    num = min(len(imgs), img_nums)
+
+    cfg = DeployConfig(args.cfg)
+    pred_cfg = PredictConfig(cfg.model, cfg.params)
+    pred_cfg.enable_use_gpu(100, 0)
+    if not args.print_detail:
+        pred_cfg.disable_glog_info()
+    pred_cfg.collect_shape_range_info(args.auto_tuned_shape_file)
+
+    predictor = create_predictor(pred_cfg)
+    input_names = predictor.get_input_names()
+    input_handle = predictor.get_input_handle(input_names[0])
+
+    for i in range(0, num):
+        data = np.array([cfg.transforms(imgs[i])[0]])
+        input_handle.reshape(data.shape)
+        input_handle.copy_from_cpu(data)
+        try:
+            predictor.run()
+        except:
+            logger.info(
+                "Auto tune fail. Usually, the error is out of GPU memory, "
```

Comment:
Auto tune failed. Usually this is caused by an OOM error on GPUs. Please check if the model or image is too large.